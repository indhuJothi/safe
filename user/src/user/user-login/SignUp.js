import React from "react";
import "./LogReg.css";
import data from "../../resources/user.json";
import { Redirect, withRouter } from "react-router-dom";
import Header from "../../common/header/Header";
import SweetAlert from "react-bootstrap-sweetalert";
import baseURL from '../../service/api';


class SignUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: "",
      email: "",
      mobile: "",
      password: " ",
      confirmPassword: "",
      userNameErr: "",
      emailErr: "",
      passErr: "",
      mobileerr: "",
      confirmpassErr: "",
      login: true,
      alert: null,
      redirectLogin: false,
      userAlreadyExsist:null
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleLogin = this.handleLogin.bind(this);
  }

  handleChange(event) {
    event.preventDefault();
    const userName = event.target.userName;
    const mobile = event.target.name;
    const password = event.target.password;
    const email = event.target.email;
    const confirmPassword = event.target.confirmPassword;
    this.setState({
      [userName]: event.target.value,
      [email]: event.target.value,
      [mobile]: event.target.value,
      [password]: event.target.value,
      [confirmPassword]: event.target.value,
    });
  }
  handleLogin(e) {
    e.preventDefault();
    this.setState({
      redirectLogin: true,
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    let emailRes = true;
    let mobileRes = true;
    let passwordRes = true;
    let confirmPassRes = true;
    let userNameres = true;
    let nameRegex = /^[a-zA-Z\s]{3,15}$/;
    let emailRegex = /^[a-zA-Z0-9+_.-]+@[a-zA-Z.]+$/;
    let mobileRegex = /^[6-9]\d{9}$/;
    let passRegex = /^[A-Za-z0-9@\s]{3,15}$/;
    if (this.state.userName === "") {
      userNameres = true;
      this.setState({
        userNameErr: "Please Enter Your User Name",
      });
    } else {
      if (nameRegex.test(this.state.name)) {
        userNameres = false;
        this.setState({
          userNameErr: " ",
        });
      } else {
        userNameres = true;
        this.setState({
          userNameErr: "User Name Must Be between 3-15 Charachters",
        });
      }
    }

    if (emailRegex.test(this.state.email)) {
      emailRes = false;
      this.setState({
        emailErr: " ",
      });
    } else {
      emailRes = true;
      this.setState({
        emailErr: "Enter a valid email",
      });
    }
    if (mobileRegex.test(this.state.mobile)) {
      mobileRes = false;
      this.setState({
        mobileerr: " ",
      });
    } else {
      mobileRes = true;
      this.setState({
        mobileerr: "Enter a valid mobile",
      });
    }
    if (passRegex.test(this.state.password)) {
      passwordRes = false;
      this.setState({
        passErr: " ",
      });
    } else {
      passwordRes = true;
      this.setState({
        passErr: "Enter a valid password",
      });
    }
    if (this.state.confirmPassword === " ") {
      confirmPassRes = true;
      this.setState({
        confirmpassErr: "Please Enter Your Password",
      });
    } else if (this.state.password === this.state.confirmPassword) {
      confirmPassRes = false;
      this.setState({
        confirmpassErr: " ",
      });
    } else {
      confirmPassRes = true;
      this.setState({
        confirmpassErr: "password are not same",
      });
    }

    if (
      (userNameres ||
        emailRes ||
        mobileRes ||
        passwordRes ||
        confirmPassRes) === false
    ) {
      const getAlert = () => (
        <SweetAlert success title="!" onConfirm={() => this.hideAlert()}>
          You are signed in successfully
          <p>You can now Login</p>
        </SweetAlert>
      );
      const userAlert = () => (
        <SweetAlert success title="!" onConfirm={() => this.hideAlert()}>
          User Already Exisit
          <p>You can Login</p>
        </SweetAlert>
      );
      const newUser ={
        name:this.state.userName,
        email:this.state.email,
        mobile:this.state.mobile,
        password:this.state.password
        }
        this.registerUser(newUser)
        .then(response =>{
              if(response.data=="User alredy exsit..")
          {
            this.setState({
              alert:null,
              userAlreadyExsist:userAlert()
            })
         }
         else{
          this.setState({
            alert: getAlert(),
          });

         }
        })
    }
  }
  hideAlert() {
    this.setState({
      alert: null,
      login:false
    });
   
  }


  registerUser(newUserDetails){
    let apiUrl = 'http://localhost:5000/users/register'
    return baseURL.post(apiUrl,newUserDetails,{
        headers:{
            'Content-Type': 'application/json'
        }
    })
}

  render() {
    return (
      <div>
        <Header />
        <form onSubmit={this.handleSubmit}>
          <div className="base-container">
            <div class="MainContainer center">
              <button onClick={this.handleLogin} class="button">
                Login
              </button>
              <button class="button">
                Signup
              </button>
              <div className="formheader">Signup</div>
              <div className="form">
                <div>
                <p class="error">{this.state.userAlreadyExsist}</p>
                  <label htmlFor="email">User Name</label>
                  <input
                    type="text"
                    name="userName"
                    placeholder="User Name"
                    onChange={this.handleChange}
                  />
                  <div class="error">{this.state.userNameErr}</div>
                </div>
                <div>
                  <label htmlFor="email">Email</label>
                  <input
                    type="text"
                    name="email"
                    placeholder="xyz@domain.com"
                    onChange={this.handleChange}
                  />
                  <div class="error">{this.state.emailErr}</div>
                </div>
                <div>
                  <label htmlFor="Mobile">Mobile</label>
                  <input
                    type="text"
                    name="mobile"
                    placeholder="MobileNo"
                    onChange={this.handleChange}
                  />
                  <div class="error">{this.state.mobileerr}</div>
                </div>
                <div>
                  <label htmlFor="password">Password</label>
                  <input
                    type="password"
                    name="password"
                    placeholder="Password"
                    onChange={this.handleChange}
                  />
                  <div class="error">{this.state.passErr}</div>
                </div>
                <div>
                  <label htmlFor="confirmPassword">Confirm Password</label>
                  <input
                    type="password"
                    name="confirmPassword"
                    placeholder="confirm password"
                    onChange={this.handleChange}
                  />
                  <div class="error">{this.state.confirmpassErr}</div>
                </div>
              </div>
              <div>
                <input type="submit" class="submitbtn"></input>
              </div>
            </div>
          </div>
        </form>
        {this.state.alert}
        {this.state.redirectLogin ? <Redirect to="/login"></Redirect> : null}
        {this.state.login ? null : <Redirect to="/login"></Redirect>}
      </div>
    );
  }
}

export default withRouter(SignUp);
