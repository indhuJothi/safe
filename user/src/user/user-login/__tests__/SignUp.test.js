import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import SignUp from "../SignUp";
import userdata from "../user.json";

afterEach(cleanup);

describe("testing signup component", () => {
  test("rendering signup", () => {
    render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>
    );
    const signUpComponent = screen.getByTestId("signUp");
    expect(signUpComponent).toBeInTheDocument();
  });

  test("checking the given fields in the document", () => {
    render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>
    );
    const nameField = screen.getByPlaceholderText("User Name");
    const emailField = screen.getByPlaceholderText("xyz@domain.com");
    const mobileField = screen.getByPlaceholderText("MobileNo");
    const passwordField = screen.getByPlaceholderText("Password");
    const confirmPasswordField =
      screen.getByPlaceholderText("Confirm Password");

    expect(nameField).toBeInTheDocument();
    expect(emailField).toBeInTheDocument();
    expect(mobileField).toBeInTheDocument();
    expect(passwordField).toBeInTheDocument();
    expect(confirmPasswordField).toBeInTheDocument();
  });

  test("checking field value matchin the condtion", () => {
    render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>
    );

    const nameValue = screen.getByPlaceholderText("User Name");
    const emailValue = screen.getByPlaceholderText("xyz@domain.com");
    const mobileValue = screen.getByPlaceholderText("MobileNo");
    const passwordValue = screen.getByPlaceholderText("Password");
    const confirmPasswordValue =
      screen.getByPlaceholderText("Confirm Password");

    fireEvent.submit(nameValue, { target: { value: "Indhu" } });
    fireEvent.submit(emailValue, { target: { value: "indhu99.p@gmail.com" } });
    fireEvent.submit(mobileValue, { target: { value: 7339467878 } });
    fireEvent.submit(passwordValue, { target: { value: "Indhu@99" } });
    fireEvent.submit(confirmPasswordValue, { target: { value: "Indhu@99" } });
    expect(nameValue.value).toMatch(/^[a-zA-Z\s]{3,15}$/);
    expect(emailValue.value).toMatch(/^[a-zA-Z0-9+_.-]+@[a-zA-Z.]+$/);
    expect(mobileValue.value).toMatch(/^[6-9]\d{9}$/);
    expect(passwordValue.value).toMatch(/^[A-Za-z0-9@\s]{3,15}$/);
    expect(confirmPasswordValue.value).toMatch(passwordValue.value);
  });

  test("checking field values on fireEvent", () => {
    render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>
    );

    const nameValue = screen.getByPlaceholderText("User Name");
    const emailValue = screen.getByPlaceholderText("xyz@domain.com");
    const mobileValue = screen.getByPlaceholderText("MobileNo");
    const passwordValue = screen.getByPlaceholderText("Password");
    const confirmPasswordValue =
      screen.getByPlaceholderText("Confirm Password");

    expect(nameValue.value).toBe("");
    expect(emailValue.value).toBe("");
    expect(mobileValue.value).toBe("");
    expect(passwordValue.value).toBe("");
    expect(confirmPasswordValue.value).toBe("");
    fireEvent.submit(nameValue, { target: { value: "Indhu" } });
    expect(nameValue.value).toBe(userdata.user.Name);
    fireEvent.submit(emailValue, { target: { value: "indhu@gmail.com" } });
    expect(emailValue.value).toBe(userdata.user.Email);
    fireEvent.submit(mobileValue, { target: { value: 7339467878 } });
    expect(mobileValue.value).toBe(userdata.user.mobile);
    fireEvent.submit(passwordValue, { target: { value: "Indhu@99" } });
    expect(passwordValue.value).toBe(userdata.user.password);
    fireEvent.submit(confirmPasswordValue, { target: { value: "Indhu@99" } });
    expect(confirmPasswordValue.value).toBe(userdata.user.ConfirmPassword);
  });

  test("get snapshot", () => {
    render(
      <MemoryRouter>
        <SignUp />
      </MemoryRouter>
    );
    const tree = renderer
      .create(
        <MemoryRouter>
          <SignUp />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
