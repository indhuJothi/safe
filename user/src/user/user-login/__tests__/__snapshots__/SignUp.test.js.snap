// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing signup component get snapshot 1`] = `
<div
  data-testid="signUp"
>
  <div>
    <div
      className="body"
    >
      <div
        className="header"
      >
        <span
          className="apptitle"
        >
          Bus Booking App
        </span>
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="base-container"
    >
      <div
        className="MainContainer center"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          Login
        </button>
        <button
          className="button"
        >
          Signup
        </button>
        <div
          className="formheader"
        >
          Signup
        </div>
        <div
          className="form"
        >
          <div>
            <p
              className="error"
            />
            <label
              htmlFor="email"
            >
              User Name
            </label>
            <input
              name="userName"
              onChange={[Function]}
              placeholder="User Name"
              type="text"
            />
            <div
              className="error"
            >
              
            </div>
          </div>
          <div>
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              name="email"
              onChange={[Function]}
              placeholder="xyz@domain.com"
              type="text"
            />
            <div
              className="error"
            >
              
            </div>
          </div>
          <div>
            <label
              htmlFor="Mobile"
            >
              Mobile
            </label>
            <input
              name="mobile"
              onChange={[Function]}
              placeholder="MobileNo"
              type="text"
            />
            <div
              className="error"
            >
              
            </div>
          </div>
          <div>
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <div
              className="error"
            >
              
            </div>
          </div>
          <div>
            <label
              htmlFor="confirmPassword"
            >
              Confirm Password
            </label>
            <input
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
            />
            <div
              className="error"
            >
              
            </div>
          </div>
        </div>
        <div>
          <input
            className="submitbtn"
            type="submit"
          />
        </div>
        <br />
        <div>
           Already have an account?
          <a
            href="/login"
          >
             Login here
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;
